onecodex/viz/_heatmap.py

```diff
@@ -238,6 +251,24 @@ def plot_heatmap(
                 haxis, header=alt.Header(titleOrient="bottom", labelOrient="bottom")
             )
 
+        # Drop all-NaN rows, convert remaining NaNs to 0s, and concat
+        # dropped all-NaN rows to the end
+        dropped = []
+        for classification_id in all_nan_classification_ids:
```

Comment:
Since we know which rows we want to move to the end, I think we can accomplish this with less work:

```python
ids_with_data = []
for classification_id in df.index:
    if classification_id not in all_nan_classification_ids:
        ids_with_data.append(classification_id)

df = df.reindex(ids_with_data + all_nan_classification_ids)
```

Reply:
We also need to fill the df with 0s for the `None`s that _don't_ correspond to classifications with no abundances calculated. Also, I think we would still need to replace `0.0` with `np.nan` in the case that we are dealing with a `OneCodexAccessor`

Reply:
I'm confused why a classification in `all_nan_classification_ids` has zeros instead of nans (the "all nan" part of the name is kind of misleading). Is there a way to handle that upstream from here so that `all_nan_classification_ids` will always point to classifications that are all nans?